app-frontend/src/store/reducers/link.ts

```diff
@@ -0,0 +1,104 @@
+import initialState from '../initialState';
+import { Link } from '../../types/link';
+
+enum ActionTypes {
+  LOAD_LINK_DATA = 'LOAD_LINK_DATA',
+  LOAD_LINK_FAILED = 'LOAD_LINK_FAILED',
+  LOAD_LINK_SUCCESS = 'LOAD_LINK_SUCCESS',
+  EDIT_LINK_DATA = 'EDIT_LINK_DATA',
+  EDIT_LINK_FAILED = 'EDIT_LINK_FAILED',
+  EDIT_LINK_SUCCESS = 'EDIT_LINK_SUCCESS',
+  ADD_LINK_DATA = 'ADD_LINK_DATA',
+  ADD_LINK_SUCCESS = 'ADD_LINK_SUCCESS',
+  ADD_LINK_FAILED = 'ADD_LINK_FAILED',
+}
+
+interface LinkState {
+  data: Link,
+  loading: boolean,
+  status: null | string,
```

Comment:
what is the reason for storing status here?
We usually don't store payload in store